homeassistant/components/light/__init__.py

```diff
@@ -182,7 +186,8 @@ def async_turn_on(hass, entity_id=None, transition=None, brightness=None,
         ] if value is not None
     }
 
-    hass.async_add_job(hass.services.async_call(DOMAIN, SERVICE_TURN_ON, data))
+    return hass.async_add_job(hass.services.async_call(
```

Comment:
This is not good. A callback method should not return coroutine objects to await. We used to do it and it is a) confusing and b) leads to bugs. It also breaks with every other such service helper methods all over the code base. Please revert this change.

It looks like `light.group` will have to keep track of which `turn_on` keywords map to `turn_on` attributes and have tests in place to make sure that they are not broken.

The fact that a platform would cause such a major change to a component is a good red flag that it's not a good idea.

Reply:
Please revert all changes to `light/__init__.py`

Reply:
> The fact that a platform would cause such a major change to a component is a good red flag that it's not a good idea.

That's also why I asked [the question](https://github.com/home-assistant/home-assistant/pull/12229#discussion_r171378087). But I misunderstood your answer 😅

Done, except for the `async_togglè` and the `group` import that I found no way around.

Reply:
Let's drop `async_toggle`. I don't even know if we should keep the top level functions to call services. Like, who uses them?

Reply:
Also, since there hasn't been a need for it, let's remove it. We should only add code that is going to be used.